ipn/ipnlocal/local.go

```diff
@@ -2470,6 +2457,19 @@ func (b *LocalBackend) Start(opts ipn.Options) error {
 	}
 	b.stateMachineLockedOnEntry(unlock)
 
+	// initialize Taildrive shares from saved state
+	fs, ok := b.sys.DriveForRemote.GetOK()
+	if ok {
+		currentShares := b.pm.prefs.DriveShares()
+		if currentShares.Len() > 0 {
+			var shares []*drive.Share
+			for _, share := range currentShares.All() {
+				shares = append(shares, share.AsStruct())
+			}
+			fs.SetShares(shares)
+		}
```

Comment:
We should probably stop sharing previously shared drives after switching to a profile that has no shares enabled.
Remove `if`?
```suggestion
		var shares []*drive.Share
		for _, share := range currentShares.All() {
			shares = append(shares, share.AsStruct())
		}
		fs.SetShares(shares)
```

Alternatively, update `(*LocalBackend).resetForProfileChangeLockedOnEntry()` to call `fs.SetShares(nil)` explicitly.